app/src/main/java/com/example/androidapp/ui/report/ReportFragment.java

```diff
@@ -9,26 +13,43 @@
 import androidx.annotation.NonNull;
 import androidx.fragment.app.Fragment;
 
+import com.example.androidapp.DatabaseController;
+import com.example.androidapp.MainActivity;
+import com.example.androidapp.R;
 import com.example.androidapp.databinding.FragmentReportBinding;
+import com.example.androidapp.sensors.SensorController;
 
-
-public class ReportFragment extends Fragment {
+import java.text.SimpleDateFormat;
+import java.util.Date;
 
 
+public class ReportFragment extends Fragment {
     private FragmentReportBinding binding;
+    private static TextView  steps_taskView;
 
     public View onCreateView(@NonNull LayoutInflater inflater,
                              ViewGroup container, Bundle savedInstanceState) {
-
         binding = FragmentReportBinding.inflate(inflater, container, false);
         View root = binding.getRoot();
 
+        if(MainActivity.step_sensor==null){
+            MainActivity.step_sensor=new SensorController(getContext());
+        }
+        steps_taskView = (TextView) root.findViewById(R.id.numberTodayStep);
+        steps_taskView.setText(String.valueOf(MainActivity.step_sensor.dailySteps(getContext())));
```

Comment:
Make getDailySteps() static, because it seems really wrong to instantiate all the sensors just to make a query to the db